src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/BasePlatformBehavior.shared.cs

```diff
@@ -0,0 +1,75 @@
+#if IOS || MACOS || MACCATALYST
+using PlatformView = UIKit.UIView;
+#elif ANDROID
+using PlatformView = Android.Views.View;
+#elif WINDOWS
+using PlatformView = Microsoft.UI.Xaml.FrameworkElement;
+#elif TIZEN
+using PlatformView = Tizen.NUI.BaseComponents.View;
+#elif NET6_0_OR_GREATER || (NETSTANDARD || !PLATFORM)
+using PlatformView = System.Object;
+#endif
+
+using System.ComponentModel;
+
+namespace CommunityToolkit.Maui.Behaviors;
+
+
+/// <summary>
+/// Abstract class for our behaviors to inherit.
+/// </summary>
+/// <typeparam name="TView">The <see cref="VisualElement"/> that the behavior can be applied to</typeparam>
+public abstract partial class BasePlatformBehavior<TView> : BasePlatformBehavior<TView, PlatformView>
+	where TView : Element
+{
+
+}
+
+/// <summary>
+/// Abstract class for our behaviors to inherit.
+/// </summary>
+/// <typeparam name="TView">The <see cref="VisualElement"/> that the behavior can be applied to</typeparam>
+/// <typeparam name="TPlatformView">The <see langword="class"/> that the behavior can be applied to</typeparam>
+public abstract partial class BasePlatformBehavior<TView, TPlatformView> : PlatformBehavior<TView, TPlatformView>, ICommunityToolkitBehavior<TView>
+	where TView : Element
+	where TPlatformView : class
+{
+	/// <summary>
+	/// View used by the Behavior
+	/// </summary>
+	protected TView? View { get; set; }
+
+	TView? ICommunityToolkitBehavior<TView>.View
```

Comment:
`ICommunityToolkitBehavior` interface is `public`, this will allows any external member to set the `View` value when cast it to the `interface`, is it intentional?

Reply:
It's a `protected` interface property! 

Since it's marked `protected`, devs won't be able to manually set its value for any of our behaviors from a different class. Here's a screenshot of the compiler error I get when trying to change `ICommunityToolkitBehavior.View` in the IDE:

<img width="1133" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/13558917/28106eb4-7fdb-4e3e-b6de-18529b6e8da0">

Reply:
And here's another screenshot from the same code showing that the variable `behavior` is cast to `ICommunityToolkitBehavior`:
<img width="1127" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/13558917/55afe590-91f0-4364-beb2-4c38dbe90889">